### ISSUE TITLE
Name equivalence

### ISSUE THREAD

**Issue by fbaldner** [al-caughey/Google-Meet-Attendance#47]
Would it be possible to create a feature in which I would create an equivalence between two names?
For instance, in the attached picture:
![chrome_pqDF9uexvj](https://user-images.githubusercontent.com/35616989/91637794-54552100-e9e1-11ea-832c-0b8f4d56575c.png)
The two students are the same. But my college's system didn't handle names with accents properly and it added those characters. But the two lines are, essentially, the same student.
There's also  the case in which I add their whole names, but they use their personal gmail accounts, which will be different, like this case:
![chrome_uxHZXr3sDd](https://user-images.githubusercontent.com/35616989/91637841-adbd5000-e9e1-11ea-9067-6f46cbc89052.png)
It's the same person, but the bottom one is his name from the school's system (full name) and the top one is his name from his personal google account.

**Comment by AlCaughey** [comment]
Yes, it is possible.  The question is how best to do that?
Can you send the file to me so that I can investigate?

**Comment by Eagle-Man** [comment]
Google Meet Attendance needs to match the display names, so in the second case, you would need to edit the full name to match their Google display name, or ask them to change their display name.  If you have them listed in the order of your roster, then you can easily match up the names even if the display names are not the same.

@AlCaughey I'm not sure the second case is one you would want to adapt to.  You could check to see if the display name matches part of the listed name, but then you could run into an issue where "Jack Law" and "Jack Lawson" would both match the latter.  In the case of the first, I think it's a case of converting 2 ASCII characters into one Unicode character, probably by identifying the first character and then merging the hex codes of that and the next character into one hex code (or the other way around).

**Comment by fbaldner** [comment]
> Yes, it is possible. The question is how best to do that? Can you send the file to me so that I can investigate?

Sure I can! here it is! I renamed it to txt because github didn't accept html.
[Circuitos Elétricos 2 (2020-08-28) - Copy.txt](https://github.com/al-caughey/Google-Meet-Attendance/files/5145998/Circuitos.Eletricos.2.2020-08-28.-.Copy.txt)

As far as the second case is concerned, I think the issue that @Eagle-Man stated (Jack Law vs Jack Lawson) is a serious problem indeed. But what I had in mind was something more manual for the teacher.

I have tried asking them time and time again to use their university accounts to which I get all sorts of responses, from "I'm already using" (which is clearly false) to "Ah, ok, I'll change" and he'll return with the same account. As easy as it might be, I grew tired of asking and if there is an option to simply say that name_a = name_b, that'd be awesome.

I don't know if creating a variable for every student would be too cumbersome, and in this variable there would be a property called "aliases", similar to how python has dicts. And there I'd configure to recognize alternate names.

**Comment by Eagle-Man** [comment]
Ah, like a username dictionary of xName matches yName.  Interesting concept.

**Comment by AlCaughey** [comment]
What I'm playing with now is an alternate name beside the display name e.g.,
<display-name> (<allowed-alternate(s)>)
Billy Smith Jones (Billy Smith-Jones)... so that the name sorts properly if
sort-order=='last'
Billy Smith Jones (billys_parent@email.address)... in case they log-in via
parent credentials

Or something like that..

**Comment by fbaldner** [comment]
Just had to leave a note here saying that, so far, its working! I just got into class, saw that there was a new version! So I gave the students a couple of problems to work on and started testing it right away!!
![image](https://user-images.githubusercontent.com/35616989/92164227-65e85f80-ee0b-11ea-9f05-681da5a1c3dc.png)
As soon as this class ends I'll check the report, and I'll test in the other three classes I have today.

**Comment by AlCaughey** [comment]
NB - There's a bug in v0.9.0. A fix has already been posted to the web
store.

Don't worry, all of your attendance data is still in the file(s), it is
just hidden by a rogue line of CSS code.  For a more detailed description
and an explanation on how you can fix the problem yourself, please see
https://youtu.be/XqtmeSwnxu8

I'm terribly sorry for the inconvenience

Al

**Comment by fbaldner** [comment]
@AlCaughey don't worry, I saw the video!
Should I close this thread now that the issue was added to the extension?

**Comment by AlCaughey** [comment]
Sure, you can close it.

NB - in v0.9.0 there was another bug in the html reports related to the
aliasing but I didn't mention it because very few people know about it.

I have tested this new feature as best I can without actual classes.  There
could be issues still. You can probably break things quite easily... eg,
ambiguous matches will likely generate interesting results in the reports

**Comment by fbaldner** [comment]
I just looked at my reports and it didn't really show the attendance of the students which used the alias.
I looked in  the html and the attendance was recorded, just not shown. Did you solve this bug or do you need an example in order to test things out?

**Comment by AlCaughey** [comment]
Can you send me the file? I suspect that I know what's going wrong. But it
would be good to confirm

**Comment by fbaldner** [comment]
Sure can! These are from my three classes last night.
[Eletromagnetismo (2020-09-03).html.txt](https://github.com/al-caughey/Google-Meet-Attendance/files/5174785/Eletromagnetismo.2020-09-03.html.txt)
[Circuitos Elétricos 1 (2020-09-03).html.txt](https://github.com/al-caughey/Google-Meet-Attendance/files/5174786/Circuitos.Eletricos.1.2020-09-03.html.txt)
[Circuitos Elétricos 2 (2020-09-03).html.txt](https://github.com/al-caughey/Google-Meet-Attendance/files/5174787/Circuitos.Eletricos.2.2020-09-03.html.txt)
Added a .txt extension just to be able to upload.

**Comment by AlCaughey** [comment]
in the htm file, replace
data-student-name=
with
data-student-name*=

Note the asterisk before the equal sign...

fixed file attached

**Comment by fbaldner** [comment]
I didn't see the attached files.
Should I replace every occurrence of "data-student-name=" for "data-student-name*="?
There are 8 matches.

**Comment by AlCaughey** [comment]
Yes, that error has been fixed in v0.9.1 but does not fix the issue in the
HTML files which were already saved
